pkgs/by-name/we/wemeet/package.nix

```diff
@@ -0,0 +1,259 @@
+      hash = "sha256-avN+PHKKC58lMC5wd0yVLD0Ct7sbb4BtXjovish0ULU=";
+    };
+  };
+
+  nativeBuildInputs = [
+    dpkg
+    autoPatchelfHook
+    makeWrapper
+  ];
+
+  buildInputs = [
+    nss
+    xorg.libX11
+    xorg.libSM
+    xorg.libICE
+    xorg.libXtst
+    desktop-file-utils
+    libpulseaudio
+    libgcrypt
+    dbus
+    systemd
+    udev
+    libGL
+    libGL
+    fontconfig
+    freetype
+    openssl
+    wayland
+    libdrm
+    harfbuzz
+    openldap
+    curl
+    nghttp2
+    libunwind
+    alsa-lib
+    libidn2
+    rtmpdump
+    libpsl
+    libkrb5
+    xkeyboard_config
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    cp -r opt/wemeet $out
+    cp -r usr/* $out
+    rm $out/lib/libcurl.so
+    substituteInPlace $out/share/applications/wemeetapp.desktop \
+      --replace-fail "/opt/wemeet/wemeetapp.sh" "wemeetapp" \
+      --replace-fail "/opt/wemeet/wemeet.svg" "wemeet"
+    sed -i "s|^Prefix.*|Prefix = $out/lib|" $out/bin/qt.conf
+    mv $out/icons $out/share/icons
+    install -Dm0644 $out/wemeet.svg $out/share/icons/hicolor/scalable/apps/wemeet.svg
+    ln -s $out/bin/raw/xcast.conf $out/bin/xcast.conf
+    ln -s $out/plugins $out/lib/plugins
+    ln -s $out/resources $out/lib/resources
+    mkdir -p $out/lib/translations
+    ln -s $out/translations/qtwebengine_locales $out/lib/translations/qtwebengine_locales
+
+    runHook postInstall
+  '';
+
+  preFixup = ''
+    wrapProgram $out/bin/wemeetapp \
+      --set XDG_SESSION_TYPE "x11" \
+      --set EGL_PLATFORM "x11" \
+      --set LP_NUM_THREADS "2" \
+      --set QT_QPA_PLATFORM "xcb" \
+      --set QT_STYLE_OVERRIDE "fusion" \
+      --set QT_AUTO_SCREEN_SCALE_FACTOR "1" \
+      --set IBUS_USE_PORTAL "1" \
+      --unset WAYLAND_DISPLAY \
+      --set TZ "Asia/Shanghai" \
+      --set XKB_CONFIG_ROOT "${xkeyboard_config}/share/X11/xkb" \
+      --set LC_ALL "zh_CN.UTF-8" \
+      --prefix LD_LIBRARY_PATH : "$out/x11-wayland/1050/lib/${
+        selectSystem {
+          x86_64-linux = "x86_64-linux-gnu";
+          aarch64-linux = "aarch64-linux-gnu";
+        }
+      }:$out/lib:$out/plugins:$out/resources:$out/translations:${xorg.libXext}/lib:${xorg.libXdamage}/lib:${opencv}/lib:${xorg.libXrandr}/lib" \
+      --prefix PATH : "$out/bin" \
+      --prefix QT_PLUGIN_PATH : "$out/plugins" \
+      --run "mkdir -p \$HOME/.local/share/wemeetapp" \
+      --prefix LD_PRELOAD : "${libwemeetwrap}/lib/libwemeetwrap.so:${wemeet-wayland-screenshare}/lib/wemeet/libhook.so" \
+      --set USER_RUN_DIR "/run/user/$(id -u)" \
+      --set FONTCONFIG_DIR "$CONFIG_DIR/fontconfig" \
+      --set KDE_GLOBALS_FILE "$CONFIG_DIR/kdeglobals" \
+      --set LD_PRELOAD_WRAP "$LD_PRELOAD:${libwemeetwrap}/lib/libwemeetwrap.so" \
+      --set CONFIG_DIR "''${XDG_CONFIG_HOME:-$HOME/.config}" \
+      --set KDE_ICON_CACHE_FILE "''${XDG_CACHE_HOME:-$HOME/.cache}/icon-cache.kcache" \
+      --set WEMEET_APP_DIR "''${XDG_DATA_HOME:-$HOME/.local/share}/wemeetapp"
```

Comment:
I'm... afraid I need a by-line explanation as to what in the F is this.

Surely not all of this is necessary.

Most of this isn't necessary.

This is insanity.

Complete insanity.

Reply:
Charitably assuming that this is an extremely cursed piece of proprietary software that assumes a *lot* about its environment (and judging that this is Tencent software, it by default is), there's still *so* many things that assume things that don't exist or should never be necessary to specify:

```
      --set XDG_SESSION_TYPE "x11" \
      --set EGL_PLATFORM "x11" \
```
This shouldn't be necessary, unless the program will self-implode when a Wayland environment is detected. In which case, I wonder if it should be in Nixpkgs at all. I believe just setting `QT_QPA_PLATFORM=xcb` should be enough (unsetting `WAYLAND_DISPLAY` shouldn't be necessary either)

Reply:
I think you should exercise rational thinking, and come to the conclusion that the upstream is spouting nonsense that likely had only worked for a single person ever on this green Earth.

Reply:
I do understand that the program might have problems under Wayland (essentially nobody uses Wayland at Tencent unfortunately, so that is to be expected). But everything else is completely incoherent and superfluous to a laughable degree.

Reply:
See, it's your job to scrutinize this script and recognize what is actually necessary (e.g. needing to source an `x11-ext.sh` script for it to work on XWayland), and what's nonsense (e.g. setting the locale and timezone). One should not trust Tencent in making the best call for Linux users, and you should know that.

Reply:
Yes, good. Now ask yourself — why were they necessary, and are they *still* necessary. Do some thinking instead of just reporting back on what they do (although that is appreciated)

Reply:
Great! Now document them